src/parser.cpp

```diff
@@ -120,6 +120,13 @@ void linkFromMap(LinkObject &link, const QVariantMap &linkMap)
     QString thumbnail = linkMap.value("thumbnail").toString();
     if (thumbnail.startsWith("http"))
         link.setThumbnailUrl(QUrl(thumbnail));
+    QVariantMap media = linkMap.value("secure_media").toMap().value("oembed").toMap();
```

Comment:
this newly added code could all be in the else clause. This avoids walking the linkMap when thumbnailUrl got set

Reply:
Correct